src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,123 @@
 package core.basesyntax;
 
+import java.lang.reflect.Array;
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+import java.util.Objects;
+
 public class ArrayList<T> implements List<T> {
+    private static final int START_ARRAY_LENGTH = 10;
+    private int arraySize;
+    private T[] objects;
+
+    public ArrayList() {
+        objects = (T[]) (Array.newInstance(Object.class, START_ARRAY_LENGTH));
+    }
+
     @Override
     public void add(T value) {
-
+        if (objects.length == arraySize) {
+            objects = grow(objects);
+        }
+        objects[arraySize] = value;
+        arraySize += 1;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index != 0) {
+            boundsValidation(index - 1);
+        }
```

Comment:
index validation should be performed in one place. maybe you need separate method for this case